src/index.ts

```diff
@@ -149,9 +170,24 @@ function findTargetPath(pathMatches) {
   return operation;
 }
 
-class Oas {
-  constructor(oas, user) {
-    this.api = oas || {};
+export default class Oas {
+  api: RMOAS.OASDocument;
+
+  user: RMOAS.User;
+
+  // @todo These are always `Promise.resolve` and `Promise.reject`. Make these types more specific than `any`.
+  _promises: Array<{
+    resolve: any;
+    reject: any;
+  }>;
+
+  _dereferencing: {
```

Comment:
Would love to make these `private` properties but we have some tests that's looking at this data and I don't want to add a method to expose it just for that test.